src/portable/portable.c

```diff
@@ -951,11 +951,68 @@ static int make_marker_text(const char *image_path, OrderedHashmap *extension_im
         return 0;
 }
 
+/* 'output=' is always appended to, even if there are ENOMEM errors */
+static int append_release_info_to_unit(char **output, const PortableMetadata *release, ImageClass type) {
+        static const char *const field_names[_IMAGE_CLASS_MAX][6]= {
+                 {}, /* IMAGE_MACHINE is not expected here, the assert below enforces it */
```

Comment:
just drop this line, will NULL initialized anyway.

instead maybe add `assert(!strv_isempty(field_names[type]));` below?